sources/packages/load-test/src/utils/auth/token-helpers.ts

```diff
@@ -0,0 +1,101 @@
+import http from "k6/http";
+import { AuthorizedParties } from "../sims-api/sims-api";
+import { KEY_CLOAK_TOKEN_URL } from "../../../config.env";
+import { decodeJWT } from "./jwt-helpers";
+import { TokenCacheResponse, UserPasswordCredential } from "./models";
+
+/**
+ * Default amount of time to a token be expired and be
+ * considered candidate to be renewed.
+ */
+const TOKEN_RENEWAL_SECONDS = 30;
+
+/**
+ * Caches the token already acquired.
+ */
+const tokenCache: Record<string, TokenCacheResponse> = {};
+
+/**
+ * Define if a token need to be renewed based in the jwt exp property.
+ * @param jwtExp exp token property that defines the expiration.
+ * @param maxSecondsToExpired Amount of seconds before the token expires that should be considered to renew it.
+ * For instance, if a token has 10min expiration and the parameter is defined as 30, the token will
+ * be renewed if the token was acquired 9min30s ago or if it is already expired.
+ * @returns true if the token is expired or about to expire based on the property maxSecondsToExpired.
+ */
+function needRenewJwtToken(jwtExp: number, maxSecondsToExpired = 0): boolean {
+  const jwtExpMilliseconds = jwtExp * 1000; // Convert to milliseconds.
+  const maxSecondsToExpiredMilliseconds = maxSecondsToExpired * 1000; // Convert to milliseconds.
+  const expiredDate = new Date(
+    jwtExpMilliseconds - maxSecondsToExpiredMilliseconds
+  );
+  return new Date() > expiredDate;
```

Comment:
I wondering, can we use `isAfter()` from the `date-utlis`

Reply:
It should be avoided. We can use pure javascript libraries in general but if a package has any dependency on the node it will likely not work. We are not sharing the libraries with the backend and the `load-test` folder is not inside the backend.